components/collector/src/model/parameters.py

```diff
@@ -43,15 +43,10 @@ def quote_if_needed(parameter_value: str) -> str:
             return urllib.parse.quote(parameter_value, safe="") if quote else parameter_value
 
         parameter_info = DATA_MODEL.sources[self.__source["type"]].parameters[parameter_key]
-        if parameter_info.type == "multiple_choice":
-            # If the user didn't pick any values, select the default value if any, otherwise select all values:
-            default_value = parameter_info.default_value
-            value = self.__parameters.get(parameter_key) or default_value or parameter_info.values or []
+        value = self.__parameters.get(parameter_key) or parameter_info.default_value
```

Comment:
Hmm, this is slightly unexpected; I guess the case `multiple_choice_without_defaults` would automagically be part of `or parameter_info.default_value` ?

Reply:
Not sure what you find unexpected? In any case, this code is now simpler because the datamodel itself makes sure that the default value of `MultipleChoiceParameterWithDefaults` is filled with all values if no default values are provided at instantiation.